zeth_tool/prove_cmd.cpp

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2015-2021 Clearmatics Technologies Ltd
+//
+// SPDX-License-Identifier: LGPL-3.0+
+
+#include "zeth_tool/prove_cmd.hpp"
+
+#include "libtool/tool_util.hpp"
+#include "libzeth/serialization/r1cs_variable_assignment_serialization.hpp"
+#include "zeth_tool/tool_common.hpp"
+
+namespace zethtool
+{
+
+namespace commands
+{
+
+class prove_cmd : public generic_subcommand<prove_cmd>
+{
+public:
+    using base_class = generic_subcommand<prove_cmd>;
+
+    prove_cmd(
+        const std::string &subcommand_name, const std::string &description)
+        : base_class(subcommand_name, description), num_primary_inputs(1)
+    {
+    }
+
+    template<typename ppT, typename snarkT>
+    int execute_generic(const global_options &)
+    {
+        ppT::init_public_params();
+        libff::inhibit_profiling_info = true;
+        libff::inhibit_profiling_counters = true;
+
+        typename snarkT::proving_key proving_key;
+        {
+            std::ifstream in_s = libtool::open_binary_input_file(pk_file);
+            snarkT::proving_key_read_bytes(proving_key, in_s);
+        }
+
+        libsnark::r1cs_primary_input<libff::Fr<ppT>> primary;
+        libsnark::r1cs_auxiliary_input<libff::Fr<ppT>> auxiliary;
+        {
+            std::ifstream in_s =
+                libtool::open_binary_input_file(assignment_file);
```

Comment:
Do I understand correctly that the only way for me to generate an assignment file for now is via https://github.com/clearmatics/zeth/blob/master/prover_server/prover_server.cpp#L116, and thus by running: `prover_server --assignment-output <filename>`?

Reply:
Changed the usage string.

(Yes - currently the only way to generate the assignment is via the prover_server.)

Reply:
Got it, thanks